src/promptflow/promptflow/_core/operation_context.py

```diff
@@ -133,10 +119,14 @@ def append_user_agent(self, user_agent: str):
         Args:
             user_agent (str): The user agent information to append.
         """
-        agent = self.get(self._USER_AGENT, '')
-        user_agent = user_agent.strip()
-        if user_agent not in agent:
-            self[self._USER_AGENT] = f"{agent} {user_agent}".strip()
+        agents = self.get('user_agent', '').split(' ')
+        user_agents = set(user_agent.split(' '))
+        distinct_agents = []
+        for user_agent in user_agents:
+            if user_agent not in agents and user_agent != '':
+                distinct_agents.append(user_agent)
+
```

Comment:
I understand you want to add a user agent to the existing user agent list without repeating any values. I have some suggestions for improvement.
The docstring describes the function's purpose well. However, it could provide more details about the method of adding the user agent and the logic of avoiding duplicates.